### PR TITLE
Allow nested clips to be shared between primitives where possible.

### PR DESCRIPTION
**Pull request by glennw**
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/729)

### CODE REVIEW

File: webrender/src/tiling.rs [hunk 1]
```diff
@@ -2618,9 +2619,18 @@ impl FrameBuilder {
                                 }
 
                                 // Try to create a mask if we may need to.
-                                if prim_clip_info.is_some() || !clip_info_stack.is_empty() {
+                                if !clip_info_stack.is_empty() {
```
Comment (kvark) [inline webrender/src/tiling.rs]: Why are we no longer checking for `prim_clip_info` here?

File: webrender/src/tiling.rs [hunk 2]
```diff
@@ -192,12 +192,13 @@ impl AlphaBatchHelpers for PrimitiveStore {
         let layer_index = layer_index.0 as i32;
         let global_prim_id = prim_index.0 as i32;
         let prim_address = metadata.gpu_prim_index;
-        let clip_task_key = RenderTaskKey::CacheMask(MaskCacheKey::Primitive(prim_index));
-        let clip_task_index = if metadata.clip_task.is_some() {
-            let cache_task_id = RenderTaskId::Dynamic(clip_task_key);
-            render_tasks.get_task_index(&cache_task_id, child_pass_index)
-        } else {
-            OPAQUE_TASK_INDEX
+        let clip_task_index = match metadata.clip_task {
```
Comment (kvark) [inline webrender/src/tiling.rs]: nice refactor!

File: webrender/src/tiling.rs (hunk 1 above)
Comment (glennw) [inline webrender/src/tiling.rs]: We don't need to, as the line above pushes the primitive clip item on to the clip stack, if applicable.

Comment (kvark) [inline webrender/src/tiling.rs]: Uh oh, you are right.

**Pull request merged**